### PR TITLE
docs: create the docs folder and update repo metadata

### PR DESCRIPTION
**Pull request by tisonkun**
### Task Description

This closes https://github.com/oceanbase/oceanbase/issues/1387.

cc @caifeizhi

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+* **The target audience** of this guide is OceanBase contributors, both new and experienced.
+* **The objective** of this guide is to help contributors become an expert of OceanBase, who is familiar with its design and implementation and thus is able to use it fluently in the real world as well as develop OceanBase itself deeply.
+
+## The structure of this guide
+
+At present, the guide is composed of the following parts:
+
+1. **Get started**: Setting up the development environment, build and connect to the OceanBase server, the subsections are based on an imagined newbie user journey.
+    1. Install toolchain
```
Comment (tisonkun) [inline docs/README.md:13]: Further chapters will be linked from here.

File: README.md [hunk 2]
```diff
@@ -106,7 +114,7 @@ See also [success stories](https://en.oceanbase.com/customer/home) and [Who is u
 
 # Contributing
 
-Contributions are highly appreciated. Read the [Contribute to OceanBase](https://github.com/oceanbase/oceanbase/wiki/Contribute-to-OceanBase) guide to getting started.
+Contributions are highly appreciated. Read the [development guide](docs/README.md) to getting started.
```
Comment (caifeizhi) [inline README.md:117]: Should we separate subdirectories by language, such as docs/en_us and docs/zh_cn?

Comment (tisonkun) [reply README.md:117]: After consideration, I suggest that developers can read English, and it's hard to maintain two versions of docs especially keep them up-to-date.

We may provide multilingual docs for users, but perhaps a single version of the dev guide is the best trade-off given the audience and maintain burden.

Comment (caifeizhi) [reply README.md:117]: okey, i got it

**Review by caifeizhi — APPROVED** [review]
LGTM

**Review by a1iive — APPROVED** [review]
LGTM

**Comment by tisonkun** [comment]
It seems this PR has already been merged by https://github.com/oceanbase/oceanbase/commit/60289acff5967b86e4642346594171e123c70b7d.

I'll continue the following works and please investigate why the workflow doesn't work as expected.

**Comment by LINxiansheng** [comment]
Thanks for your contribution.
We made some mistakes that we can add your author to the 60289acff5967b86e4642346594171e123c70b7d.
So we merge this PR and Announce it here.

**Pull request merged**